test/verification.test.js

```diff
@@ -180,5 +169,44 @@ test("fetchVerificationKeys()", async (t) => {
     request: testRequest,
   });
 
-  t.deepEqual(result, publicKeys);
+  t.deepEqual(result, { id: "", keys: publicKeys });
+});
+
+test("fetchVerificationKeys() - with cache", async (t) => {
+  const mockAgent = new MockAgent();
+  function fetchMock(url, opts) {
+    opts ||= {};
+    opts.dispatcher = mockAgent;
+    return fetch(url, opts);
+  }
+
+  mockAgent.disableNetConnect();
+  const mockPool = mockAgent.get("https://api.github.com");
+  mockPool
+    .intercept({
+      method: "get",
+      path: `/meta/public_keys/copilot_api`,
+    })
+    .reply(
+      304,
+      {},
+      {
+        headers: {
+          "content-type": "application/json",
+          "x-request-id": "<request-id>",
+        },
+      },
+    );
+  const testRequest = defaultRequest.defaults({
+    request: { fetch: fetchMock },
+  });
+
+  const cache = { id: "etag-value", keys: publicKeys };
+
+  const result = await fetchVerificationKeys({
+    request: testRequest,
+    cache,
+  });
+
+  t.deepEqual(result, cache);
```

Comment:
can we add another test which sends the `fetchVerificationKeys()` request twice to simulate real-life behavior? The first one should respond with 200 and the body and a `etag` header. And the 2nd responds with a 304. 

I'm not sure how undici is working with mocking, defining different replies for the same request can be tricky. We can also file a follow up issue and take care of it later

Reply:
Took a stab at this in https://github.com/copilot-extensions/preview-sdk.js/pull/81/commits/fb4f95c46074eb97680f027ee7290b6513b065b4

Reply:
Let me update the `etag` to be something that looks like what we'd expect from the GitHub API - please hold!